### PR TITLE
Add proposal for productCategories.list

### PR DESCRIPTION
**Pull request by luisaterrabrito**

### CODE REVIEW

File: src/04-invoicing/ledgers.apib [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        + filter (object, optional)
+            + department_id: `080aac72-ff1a-4627-bfe3-146b6eee979c` (string, optional) - The ID of the department you wish to filter on.
+
++ Response 200 (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + data (array)
+            + (object)
+                + id: `2aa4a6a9-9ce8-4851-a9b3-26aea2ea14c4` (string)
+                + number: 123 (number)
```
Comment (steftrenson) [inline src/04-invoicing/ledgers.apib]: It would also make sense to add the department_id here

Comment (jdrieghe) [reply src/04-invoicing/ledgers.apib]: I'm not sure we really know the departmentId ?

Comment (luisaterrabrito) [reply src/04-invoicing/ledgers.apib]: it is a column in the db inv_accounts, so we can add it and filter by it

Comment (jdrieghe) [reply src/04-invoicing/ledgers.apib]: Oh, if it's already cleanly separated by department, yes, by all means, let's add it!

 :ok: by @luisabritotl on 06/03/2018 12:05:45

**Comment by jdrieghe** [comment]
This looks good to me, can we implement it like this?

**Comment by luisaterrabrito** [comment]
working on it

File: apiary.apib [hunk 2]
```diff
@@ -1785,6 +1785,28 @@ Get a list of available payment terms.
...
+
+    + Attributes (object)
+        + data (array)
+            + (object)
+                + id: `2aa4a6a9-9ce8-4851-a9b3-26aea2ea14c4` (string)
+                + name: 'Asian Flowers' (string)
+                + ledgers (array)
+                    + (object)
+                        + department_id: `2aa4a6a9-9ce8-4851-a9b3-26aea2ea14c6` (string)
+                        + ledger_account_number: 70100 (number)
```
Comment (jenssegers) [inline apiary.apib]: Can this also have leading zeros of letters in some countries? Or is it always numeric?

File: src/apiary.apib [hunk 3]
```diff
@@ -47,6 +47,8 @@ HOST: https://api.teamleader.eu
 
 :[Payment Terms](./05-invoicing/payment-terms.apib)
 
+:[ProductCategories](./04-invoicing/product-categories.apib)
```
Comment (jenssegers) [inline src/apiary.apib]: `05-invoicing`

 :ok: by @luisabritotl on 13/03/2018 09:01:50

File: apiary.apib (hunk 2 above)
Comment (jdrieghe) [reply apiary.apib]: Tagging @PieterReel for input.

**Review by jenssegers — CHANGES_REQUESTED** [review]
Invoicing was moved from `04` to `05` in the meanwhile

File: apiary.apib (hunk 2 above)
Comment (luisaterrabrito) [reply apiary.apib]: :ok: by @luisabritotl on 13/03/2018 09:01:47

**Review by jenssegers — APPROVED** [review]

**Review by jdrieghe — APPROVED** [review]
Merge after implementation is deployed

File: src/05-invoicing/product-categories.apib [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
++ Response 200 (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + data (array)
+            + (object)
+                + id: `2aa4a6a9-9ce8-4851-a9b3-26aea2ea14c4` (string)
+                + name: 'Asian Flowers' (string)
+                + ledgers (array)
+                    + (object)
+                        + department_id: `2aa4a6a9-9ce8-4851-a9b3-26aea2ea14c6` (string)
```
Comment (mrtus) [inline src/05-invoicing/product-categories.apib]: Should this be done like this @stivni @jenssegers ?
Shouldn't we use the same id format everywhere??

I'd expect this as format

```json
"department":{
    "type": "department",
    "id": "<something uuid-ish>"
}
```

Comment (jenssegers) [reply src/05-invoicing/product-categories.apib]: Yes; https://github.com/teamleadercrm/api/blob/master/GUIDELINES.md#returning-relationships

Comment (jenssegers) [reply src/05-invoicing/product-categories.apib]: Good catch.

**Review by jenssegers — CHANGES_REQUESTED** [review]
Department should be object in response.

File: src/05-invoicing/product-categories.apib (hunk 4 above)
Comment (luisaterrabrito) [reply src/05-invoicing/product-categories.apib]: thanks :D

**Review by mrtus — APPROVED** [review]
🎉 Looks better!

Good work!

**Review by jenssegers — APPROVED** [review]

**Pull request merged**